### PR TITLE
Docs updates

### LINKED ISSUE #171: Improve documentation of install steps
I have a dv vm build that needs to get and build mesos-dns.
This was working fine up until this morning.
Now when it does:
```bash
go get github.com/mesosphere/mesos-dns
```
I get:
```bash
virtualbox-ovf: failed: [127.0.0.1] => {"changed": true, "cmd": "go get github.com/mesosphere/mesos-dns", "delta": "0:03:02.253808", "end": "2015-06-12 17:55:25.287440", "rc": 2, "start": "2015-06-12 17:52:23.033632", "warnings": []}
    virtualbox-ovf: stderr: # github.com/mesos/mesos-go/mesosproto
    virtualbox-ovf: /home/vagrant/go/src/github.com/mesos/mesos-go/mesosproto/log.pb.go:663: undefined: proto.Skip
    virtualbox-ovf: /home/vagrant/go/src/github.com/mesos/mesos-go/mesosproto/log.pb.go:872: undefined: proto.Skip
    virtualbox-ovf: /home/vagrant/go/src/github.com/mesos/mesos-go/mesosproto/log.pb.go:913: undefined: proto.Skip
    virtualbox-ovf: /home/vagrant/go/src/github.com/mesos/mesos-go/mesosproto/log.pb.go:999: undefined: proto.Skip
    virtualbox-ovf: /home/vagrant/go/src/github.com/mesos/mesos-go/mesosproto/log.pb.go:1058: undefined: proto.Skip
    virtualbox-ovf: /home/vagrant/go/src/github.com/mesos/mesos-go/mesosproto/log.pb.go:1134: undefined: proto.Skip
    virtualbox-ovf: /home/vagrant/go/src/github.com/mesos/mesos-go/mesosproto/log.pb.go:1274: undefined: proto.Skip
    virtualbox-ovf: /home/vagrant/go/src/github.com/mesos/mesos-go/mesosproto/log.pb.go:1350: undefined: proto.Skip
    virtualbox-ovf: /home/vagrant/go/src/github.com/mesos/mesos-go/mesosproto/log.pb.go:1471: undefined: proto.Skip
    virtualbox-ovf: /home/vagrant/go/src/github.com/mesos/mesos-go/mesosproto/log.pb.go:1663: undefined: proto.Skip
    virtualbox-ovf: /home/vagrant/go/src/github.com/mesos/mesos-go/mesosproto/log.pb.go:1663: too many errors
```

How can i fix this??

### PR DESCRIPTION
**Pull request by tsenart**
This commit brings our docs up to date with the current state of the
project.

Fixes #171, #229, #239, #264

### CODE REVIEW

File: docs/docs/index.md [hunk 1]
```diff
@@ -1,34 +1,11 @@
...
-make all
-``` 
-
-This generates `mesos-dns`, a statically-linked binary that can be installed anywhere. You will find a sample configuration file `config.json` in the same directory. 
-
-We have built and tested Mesos-DNS with `go` versions 1.3.3 and 1.4. Newer versions of `go` should work as well. 
-
-
-### Running Mesos-DNS
+The official distribution and installation channel is pre-compiled
```
Comment (jdef) [inline docs/docs/index.md]: do we really plan to host installers via this channel? i was under the impression that go binaries would be here, but not packaging/installers.

Comment (tsenart) [inline docs/docs/index.md]: Installers? We're just talking about Go binaries.

File: docs/_layouts/docs.html [hunk 2]
```diff
@@ -39,11 +30,6 @@
           Performance Tuning
         </a>
       </li>
-      <li>
```
Comment (jdef) [inline docs/_layouts/docs.html]: why remove this?

File: docs/docs/index.md (hunk 1 above)
Comment (jdef) [inline docs/docs/index.md]: "... and installation ..."

**Comment by jdef** [comment]
overall :+1:, minor comments

File: docs/docs/index.md (hunk 1 above)
Comment (tsenart) [inline docs/docs/index.md]: You're right. Even if installation is just a `mv` to a known directory, this needs rephrasing. What do you suggest?

File: docs/_layouts/docs.html (hunk 2 above)
Comment (tsenart) [inline docs/_layouts/docs.html]: We don't support that `mesos-dns-pkg` repo anymore.

File: docs/docs/index.md (hunk 1 above)
Comment (jdef) [inline docs/docs/index.md]: Something along the lines of:
"""
Stable binaries for the project are published using via the GitHub release
channel: https://github.com/mesosphere/mesos-dns/releases.
"""

**Pull request merged**